lib/components/Snackbar/snackbar.stories.tsx

```diff
@@ -37,103 +59,165 @@ const meta = {
             description:
                 "Optional. Must be included if we have actionText to trigger action and close snackbar.",
         },
-        isOpen: {
-            description:
-                "Required. Controls opening and closing of snackbar. Should be set to false by default.",
-            control: "boolean",
-        },
-        onClose: {
-            description:
-                "Required. Needed to close snackbar after a certain time.",
-        },
         hasCloseButton: {
             control: "boolean",
             description: "Optional. Set to true by default.",
         },
     },
-} satisfies Meta<typeof Snackbar>;
-
-export default meta;
-
-type Story = StoryObj<typeof meta>;
-
-export const defaultSnackbar: Story = {
+    decorators: [
+        (Story) => (
+            <SnackbarProvider>
+                <SnackbarController />
+                <Story />
+            </SnackbarProvider>
+        ),
+    ],
     parameters: {
         docs: {
             story: {
-                height: "100px",
+                height: "200px",
             },
+            page: () => (
+                <>
+                    <Markdown>
+                        {`#How to use snackbar component
+First import SnackbarProvider and SnackbarController in root of your application and wrap SnackbarController inside SnackbarProvider as shown below.
```

Comment:
It's a very interesting approach with the `useSnackbar` hook and Context provider. :)
I like the idea, however, I doubt that we really need the context or that we need to wrap the entire application in this Context provider just to use Snackbars alone.
In my opinion, the context and the hook are not really necessary to use for managing the state of snackbars in this library.
The state of the Snackbars queue should be managed outside the library - in our application - including removal, adding and other manipulations with snackbars.
So to make stories work, this state can be handled using `useState` inside stories.
`SnackbarController` can be renamed into `SnackbarsQueue` and turned into a regular an exported component that would accept `snackbars` prop as an array of objects for example, and it would notify the parent about the removal of a particular snackbar using a callback.
@balakrishna-deriv @akmal-deriv @prince-deriv  What do you guys think?

Reply:
I agree on you @maryia-deriv, we need to keep the components dumb and simple.

Reply:
@maryia-deriv @prince-deriv I asked @henry-deriv to work on this component. Considering that we are splitting the apps into standalone apps like dtrader, trader's hub, p2p, etc.., we need to abstract the logic of queuing in one place. Otherwise, we will have to reimplement the same thing in different repos.

Reply:
Snackbar is a dumb component. It can be used separately without any queuing as well.

Reply:
I'd agree that the usage of any component should be as simple as possible, which is not the exact case here, however, given the point that we are indeed moving towards multiple repos approach that are still visually a part of a single application, we will have to manage the queue in a single place, not inside the multiple apps, and I guess this is one of the better approaches for it.

Reply:
ok I understand the requirements. I think this is the only way to avoid reimplementation on each project.